packages/react-components/src/button/src/Button.tsx

```diff
@@ -1,82 +1,97 @@
 import "./TextButton.css";
 
 import { Box } from "../../box";
+import { ComponentProps, ElementType, ForwardedRef, ReactNode, SyntheticEvent, useMemo } from "react";
 import { Text } from "../../text";
-import { any, bool, elementType, func, number, oneOf, oneOfType, string } from "prop-types";
-import { createSizeAdapter, cssModule, mergeProps, omitProps, slot, useSlots } from "../../shared";
+import { createSizeAdapter, cssModule, forwardRef, mergeProps, omitProps, slot, useSlots } from "../../shared";
 import { embeddedIconSize } from "../../icons";
-import { forwardRef, useMemo } from "react";
 import { useButton } from "./useButton";
 import { useFormButton } from "../../form";
 import { useToolbarProps } from "../../toolbar";
 
-const propTypes = {
+interface InnerButtonProps {
     /**
      * The button style to use.
      */
-    variant: oneOf(["solid", "outline", "ghost"]),
+    variant?: "solid" | "outline" | "ghost";
     /**
      * The button color accent.
      */
-    color: oneOf(["primary", "secondary", "danger", "inherit"]),
+    color?: "primary" | "secondary" | "danger" | "inherit";
     /**
      * The button shape.
      */
-    shape: oneOf(["pill", "rounded", "circular"]),
+    shape?: "pill" | "rounded" | "circular";
     /**
      * Whether or not the button content should takes additional space.
      */
-    condensed: bool,
+    condensed?: boolean;
     /**
      * Whether or not the button should autoFocus on render.
      */
-    autoFocus: oneOfType([bool, number]),
+    autoFocus?: boolean | number;
     /**
      * Whether or not the button take up the width of its container.
      */
-    fluid: bool,
+    fluid?: boolean;
     /**
      * A button can show a loading indicator.
      */
-    loading: bool,
+    loading?: boolean;
     /**
      * A button can vary in size.
      */
-    size: oneOf(["sm", "md"]),
+    size?: "sm" | "md";
     /**
      * Whether or not the button is disabled.
      */
-    disabled: bool,
+    disabled?: boolean;
     /**
      * The button type.
      */
-    type: oneOf(["button", "submit", "reset"]),
+    type?: "button" | "submit" | "reset";
     /**
-     * Called when the button is click.
-     * @param {SyntheticEvent} event - React's original SyntheticEvent.
-     * @returns {void}
-     */
-    onClick: func,
+    * Called when the button is click.
+    * @param {SyntheticEvent} event - React's original SyntheticEvent.
+    * @returns {void}
+    */
+    onClick?: (event: SyntheticEvent) => void
```

Comment:
Do we want to be more specific for handlers like a button `onClick` ? We could use `MouseEvent` here.